src/lib/compiler/compileMusic.js

```diff
@@ -25,7 +27,37 @@ const compileMusic = async ({
       progress,
       warnings
     });
+    // Modify Music_Track_x.c to improve bank allocation
+    // TODO Recursive bank allocation.
+    let musicTrackTemp = await fs.readFile(`${buildRoot}/src/music/${track.dataName}.c`, "utf8");
+
+    // Approximate data size by dividing file lenth, over estimates, better than underestimate.
+    var musicSize = Math.floor(musicTrackTemp.length/5.5);
+    progress(track.dataName + ' aprox size in bytes: ' + musicSize);
+
+    if (musicSize + banksize[banksize.length-1] < 16384) {
+      // fill bank, replaces bank=8 with current bank
+      banksize[banksize.length-1] = (banksize[banksize.length-1] + musicSize);
+      musicTrackTemp = musicTrackTemp.replace('#pragma bank=8', '#pragma bank='+ (musicBanksStart+banksize.length));
```

Comment:
this line can be taken out of the `if` as it's also happening in the `else` path